bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/servlets/FormMetaDataDataSourceServlet.java

```diff
@@ -193,6 +193,14 @@ private List<Resource> getDataSourceResources(HttpServletRequest request, Resour
                     metaDataList = formMetaData.getPrefillActions();
                     resources = this.getResourceListFromComponentDescription(metaDataList, resourceResolver);
                     break;
+                case CUSTOM_PROPERTIES:
+                    if (policy != null) {
+                        List<String> customPropertyGroups = ComponentUtils.getCustomPropertyGroupsFromPolicy(policy, resourceResolver);
```

Comment:
This needs to flow via FormMetaData provider which exist in cq-guides, this will automatically enforce you to move the servlet also to cq-guides.